stdlib/gc.ml

```diff
@@ -108,8 +108,8 @@ type alarm_rec = {active : alarm; f : unit -> unit}
 
 let rec call_alarm arec =
   if Atomic.get arec.active then begin
-    finalise call_alarm arec;
     arec.f ();
+    finalise call_alarm arec;
```

Comment:
I propose something like:
```suggestion
    let finally () = finalise call_alarm arec in
    Fun.protect arec.f ~finally
```
One of the earliest documented use of Gc.alarms was to raise an exception: https://discuss.ocaml.org/t/todays-trick-memory-limits-with-gc-alarms/4431

Reply:
Well spotted!
Thank you for pointing out this oversight and the suggested fix.

Reply:
It looks like introducing a dependency here breaks the build in some way, you may need more Makefile fiddling.